integration/src/main/java/scripts/utils/MultiResourcesUtils.java

```diff
@@ -33,131 +26,72 @@ private static String makeName() {
-                      makeName(),
-                      null,
-                      CloningInstructionsEnum.NOTHING);
-              return resource.getMetadata();
-            },
-
-            // BQ data table reference
-            () -> {
-              GcpBigQueryDataTableResource resource =
-                  BqDatasetUtils.makeBigQueryDataTableReference(
-                      bqTable, referencedGcpResourceApi, workspaceId, makeName());
-              return resource.getMetadata();
-            },
-
-            // GitHub repository reference
-            () -> {
-              GitRepoResource resource =
-                  GitRepoUtils.makeGitRepoReference(
-                      gitRepo, referencedGcpResourceApi, workspaceId, makeName());
-              return resource.getMetadata();
-            });
-
-    // Build the resources
-    List<ResourceMetadata> resources = new ArrayList<>();
-    for (int i = 0; i < resourceCount; i++) {
-      int index = i % makers.size();
-      resources.add(makers.get(index).get());
-    }
-    return resources;
+    // Create a shared GCS bucket, a private GCS bucket, and a shared BQ dataset
+    GcpGcsBucketResource sharedBucket =
+        GcsBucketUtils.makeControlledGcsBucketUserShared(
+                controlledGcpResourceApi, workspaceId, makeName(), CloningInstructionsEnum.NOTHING)
+            .getGcpBucket();
+    GcpGcsBucketResource privateBucket =
+        GcsBucketUtils.makeControlledGcsBucketUserPrivate(
+                controlledGcpResourceApi, workspaceId, makeName(), CloningInstructionsEnum.NOTHING)
+            .getGcpBucket();
+    GcpBigQueryDatasetResource sharedDataset =
+        BqDatasetUtils.makeControlledBigQueryDatasetUserShared(
+            controlledGcpResourceApi,
+            workspaceId,
+            makeName(),
+            null,
+            CloningInstructionsEnum.NOTHING);
+    GcpAiNotebookInstanceResource notebook =
+        NotebookUtils.makeControlledNotebookUserPrivate(
+                workspaceId, /*instanceId=*/ null, /*location=*/ null, controlledGcpResourceApi)
+            .getAiNotebookInstance();
+    // Create references to the above buckets and datasets
+    GcpGcsBucketResource sharedBucketReference =
+        GcsBucketUtils.makeGcsBucketReference(
+            sharedBucket.getAttributes(),
+            referencedGcpResourceApi,
+            workspaceId,
+            makeName(),
+            CloningInstructionsEnum.NOTHING);
+    GcpGcsBucketResource privateBucketReference =
+        GcsBucketUtils.makeGcsBucketReference(
+            privateBucket.getAttributes(),
+            referencedGcpResourceApi,
+            workspaceId,
+            makeName(),
+            CloningInstructionsEnum.NOTHING);
+    GcpBigQueryDatasetResource datasetReference =
+        BqDatasetUtils.makeBigQueryDatasetReference(
+            sharedDataset.getAttributes(), referencedGcpResourceApi, workspaceId, makeName());
+    return List.of(
```

Comment:
You could make a map from resoruce type to metadata instead of a list.

Reply:
We could, but at least right now callers don't make any assumptions about the types resources in the returned list, which I prefer